File: accsaber-database/src/main/java/de/ixsen/accsaber/database/repositories/view/OverallAccSaberPlayerRepository.java
package de.ixsen.accsaber.database.repositories.view;

import de.ixsen.accsaber.database.repositories.ReadOnlyRepository;
import de.ixsen.accsaber.database.views.AccSaberPlayer;
import de.ixsen.accsaber.database.views.OverallAccSaberPlayer;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface OverallAccSaberPlayerRepository extends ReadOnlyRepository<OverallAccSaberPlayer, Long, AccSaberPlayer> {

    Optional<AccSaberPlayer> findPlayerByPlayerId(Long playerId);

}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/repositories/view/OverallAccSaberPlayerRepository.kt
package de.ixsen.accsaber.database.repositories.view

import de.ixsen.accsaber.database.repositories.ReadOnlyRepository
import de.ixsen.accsaber.database.views.AccSaberPlayer
import de.ixsen.accsaber.database.views.OverallAccSaberPlayer
import org.springframework.stereotype.Repository

@Repository
interface OverallAccSaberPlayerRepository : ReadOnlyRepository<OverallAccSaberPlayer, Long, AccSaberPlayer> {
    fun findPlayerByPlayerId(playerId: Long): AccSaberPlayer?
}